### DISCUSSION TITLE
What should the CIP priority of a non-real atom be? (e.g. an R group

### DISCUSSION THREAD

**Discussion by d-b-w** [rdkit/rdkit#7891] [Development]
RDKit's CIPLabeler assigns CIP priorities based on the IUPAC rules + the additions from "Algorithmic Analysis of Cahn-Ingold-Prelog Rules of Stereochemistry: Proposals for Revised Rules and a Guide for Machine Implementation" by Hansen _et al._

In these rules, I have not found discussion of non-real atoms like queries or R groups. Currently, RDKit does label centers for which a "dummy atom" is needed to break ties. I can think of 4 reasonable treatments of "dummy atoms", each has some problems.

1. Higher priority than all real elements
2. Lower priority than all real elements
3. Higher priority than hydrogen, but lower than everything else
4. If a dummy atom is needed in tie-breaking, show no label

Currently RDKit does #2, so any change in behavior should be considered carefully.

Here's a set of three scaffolds (with example R groups). Each of these approaches violates my personal assumptions on one of the examples:
<img width="828" alt="CIP_dummy_atom" src="https://github.com/user-attachments/assets/e5889dd9-5924-4c8f-9d17-4829e752a9e2">

**Comment by greglandrum** [comment]
Just to confirm/clarify: yes, in the RDKit dummy atoms have an atomic number of zero, so they have the lowest priority in the CIP code.

My two cents on your examples: I don't think it makes sense to generate CIP codes for these molecules since the whole molecule isn't specified. But I think that's something the code calling the CIP assigner should handle.

**Comment by ZontaNicola** [comment]
My preference would be to get to point 4. I agree with Greg that generating CIP codes when the dummy atom is involved is not ideal. But in a case where  dummy atoms are present but not involved in determine a stereocenter, I think we should calculate and display it, I think we've had requests for that from customers already. I'm ok for the calling code to decide whether to show or not, but having the CIP assigner return whether or not a dummy atom was involved in the determination of the labels would be super helpful

**Comment by d-b-w** [comment]
Good point - I guess that's the way that this interacts with the CIP determination code. It makes sense to want CIP labels when the whole priority can be scored without using any dummy atoms, but to skip it once a dummy atom is involved:

<img width="447" alt="labels_for_scaffold" src="https://github.com/user-attachments/assets/be0ddc3a-df0c-4338-8563-75fc741a038f">

I think that this mostly comes up for scaffolds and R-groups. When we're talking about queries, CIP labelling is just silly.

**Comment by d-b-w** [comment]
Oh - and for reference, here's the behavior of some other tools with `C[C@@H](O)C[*]`

* RDKit: _S_, because dummy is lowest priority
* Marvin (ChemAxon): _R_. It looks like dummy is highest priority (https://marvinjs-demo.chemaxon.com/latest/)
* Ketcher (Indigo): Error message. https://lifescience.opensource.epam.com/KetcherDemo/index.html?api_path=/v2
* Schrödinger (mmstereo) _S_, because dummy is lowest priority
